rts/Rendering/Units/UnitDrawer.cpp

```diff
@@ -547,18 +548,25 @@ void CUnitDrawerLegacy::DrawUnitIconsScreen() const
 			const float x1 = (pos.x + offset) / globalRendering->viewSizeX;
 			const float y1 = (pos.y - offset) / globalRendering->viewSizeY;
 
-			if (x1 < 0 && x0 > 1 && y0 < 0 && y1 > 1)
+			if (x1 < 0 || x0 > 1 || y0 < 0 || y1 > 1) // This was previously &&, if any of these are true, then the icon is not drawn.
 				continue; // don't try to draw when totally outside the screen
 
+			if (hasVisibleIcons == false) {
+				hasVisibleIcons = true;
+				icon->BindTexture();
```

Comment:
This should go to the block at https://github.com/beyond-all-reason/spring/pull/1171/files#diff-e5733cd172ae4abf67fe875d3c3a134aab2689232f077baae8dbf0ba527bea02R568

Reply:
Actually, you can get rid of `hasVisibleIcons` completely and just check `rb.ShouldSubmit(true)` here https://github.com/beyond-all-reason/spring/pull/1171/files#diff-e5733cd172ae4abf67fe875d3c3a134aab2689232f077baae8dbf0ba527bea02R567 instead